src/sentry_logger.c

```diff
@@ -43,11 +43,12 @@ sentry__logger_defaultlogger(
     const char *prefix = "[sentry] ";
     const char *priority = sentry__logger_describe(level);
 
-    char *format = sentry_malloc(
-        strlen(prefix) + strlen(priority) + strlen(message) + 1);
-    strcpy(format, prefix);
-    strcat(format, priority);
-    strcat(format, message);
+    size_t len = strlen(prefix) + strlen(priority) + strlen(message) + 1;
+    char *format = sentry_malloc(len);
+
+    strncpy(format, prefix, strlen(prefix));
+    strncat(format, priority, strlen(priority));
+    strncat(format, message, strlen(message));
```

Comment:
Where is the difference to a trivial `strcat`? Since you just use `strlen`, this should be exactly equivalent.

Reply:
This is actually quite frustrating. So msvc suggests to use `strcat_s` instead, as a save alternative. But even though that is supposed to be in C11, neither clang nor gcc have it for whatever reason. There is `strlcat` only on BSDs apparently, and `strscat` only in the linux kernel.

Reply:
Right, `strcat_s` is marked as optional in the C standard and not in the C++ standard afaik. In that case, maybe add a comment above that block explaining what the intention is.